addons/sale_mrp/models/sale.py

```diff
@@ -13,6 +14,23 @@ class SaleOrder(models.Model):
         compute='_compute_mrp_production_count',
         groups='mrp.group_mrp_user')
 
+    @api.depends('picking_ids')
+    def _compute_picking_ids(self):
+        '''Overriding to filter transfers related to sale order '''
+        super()._compute_picking_ids()
+        for order in self:
+            procurement_groups = self.env['procurement.group'].search([('sale_id', '=', order.id)])
+            mrp_procurement_group_ids = set(procurement_groups.stock_move_ids.created_production_id.procurement_group_id.ids) | \
```

Comment:
No need to convert to `set`, `recordset` works like a set

Reply:
Done